src/providers/ModalContextProvider.tsx

```diff
@@ -0,0 +1,26 @@
+import React, {useState} from 'react';
+import {AppModalState, ModalContext} from '../context/ModalContext';
+
+const AppContextProvider = ({children}) => {
+  const [modalsState, setModalState] = useState<AppModalState>({
```

Comment:
instead of defining a combined state for "sortModalJokesOpen" and "RateJokesModalOpen" you can define a state for each one. i.e 

const [rateJokeModalOpen, setRateJokeModalOpen] = useState<bool>(false)
const [sortJokesModalOpen, setSortJokesModalOpen] = useState<bool>(false)

and in the provider value prop you can give it :
value= {{
rateJokeModalOpen,
setRateJokeModalOpen,
sortJokesModalOpen,
 setSortJokesModalOpen
}}